deploy/scripts/build.sh

```diff
@@ -1,22 +1,33 @@
 cd ../../
 mkdir -p probe/build
 cd probe/build
-cmake -DBUILD_DRIVER=OFF -DPROBE_VERSION=0.1.1dev ..
+agent_libs_src="/kindling/agent-libs/"
+cflag="-DBUILD_DRIVER=OFF -DPROBE_VERSION=0.1.1dev"
+if [ -d "$agent_libs_src" ]; then
+  cflag=$cflag" -DAGENT_LIBS_SOURCE_DIR=$agent_libs_src"
+fi
+cmake $cflag ..
 make
 libKindlingPath="./src/libkindling.so"
 if [ ! -f "$libKindlingPath" ]; then
   echo "compiler libkindling failed! exit!"
 
 else
-  mkdir -p ../../collector/docker/libso &&  cp -rf ./src/libkindling.so ../../collector/docker/libso/
+  mkdir -p ../../collector/docker/libso && cp -rf ./src/libkindling.so ../../collector/docker/libso/
   cp -rf ./src/libkindling.so /usr/lib64/
   cd ../../collector/
+  go mod tidy && go mod vendor
```

Comment:
Is `go mod vendor` necessary?

Reply:
Has been removed in the current version, this will slow down the image packaging speed

Reply:
AFAIK, `go mod vendor` is helpful in two cases:
1. When there is no Internet access.
2. When some packages are not available from the proxy anymore.

Kindling has been using `GOPATH` as the pkg directory and it works well. I have not seen the need to use `go mod vendor`.

Regarding the image packaging speed, unless we maintain the `vendor` directory in the Git repository, there is no noticeable difference, IMO.

Reply:
When we compile  `GOPATH = /root/go/` ，use run_docker. sh again, path ‘/root/go’ does not exist, need to download modules again. If we do vendors, the vendor path  `/kindling/collector/vendor/` will be mapped to images with Kindling code. Compiling again will reduce the time. This is only valid for multiple partial compilations. 

It's a great help to compile multiple times when we're local.

Reply:
This scenario is friendly for local build testing, but has some concerns about being a generic scenario, so I've removed it in a new commit, which should be done by tomorrow evening at the latest

Reply:
removed 'go mod vendor'